dvc/repo/stage.py

```diff
@@ -176,7 +180,12 @@ def create(
 
                 check_stage_exists(self.repo, stage, stage.path)
 
-            self.repo.check_graph(stages={stage})
+            try:
+                self.repo.check_graph(stages={stage})
+            except OutputDuplicationError as exc:
+                # Don't include the stage currently being added.
+                exc.stages.remove(stage)
+                raise OutputDuplicationError(exc.output, exc.stages) from None
```

Comment:
Would be nice to add a test for this

Reply:
Any suggestions for where to add it?

Reply:
Probably in `tests/fun/test_add`, something like: https://github.com/iterative/dvc/blob/121d56b7391f5325e5eb6ff41448980f0ede00df/tests/func/test_add.py#L749-L753 or even just update that one and use [`match`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises) with the expected exception message

Reply:
Does the message vary if the order is different from the one in the test (i.e. you run `dvc add` first)?

Reply:
Yes it does. This PR doesn't change the error for `dvc add`, only for `dvc stage add`, so not sure if it makes sense to put it there. Had trouble finding any tests about `dvc stage add` beyond testing the cmd args.

Reply:
Ok. I understand.

From a quick look, I think I would personally add it to `tests/func/test_stage.py` like:

```py
def test_stage_add_duplicated_output(tmp_dir, dvc):
    tmp_dir.dvc_gen("foo", "foo") 
    dvc.add("foo")

    with pytest.raises(
        OutputDuplicationError,
        match="Use `dvc remove foo.dvc` to stop tracking the overlapping output."
    ):
        dvc.stage.add(name="duplicated", cmd="echo bar > foo", outs=["foo"])
```

But I am also unsure about where the logic for `Stage.create` is/should be tested cc @skshetry

Reply:
Added the test you suggested @daavoo. It makes sense to me.